src/main/java/seedu/address/logic/parser/AddTaskCommandParser.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DEADLINE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PROJECT;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.AddTaskCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.task.Task;
+
+
+
+
+/**
+ * Parses input arguments and creates a new AddTaskCommand object
+ */
+public class AddTaskCommandParser implements Parser<AddTaskCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddTaskCommand
+     * and returns an AddTaskCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddTaskCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PROJECT, PREFIX_DEADLINE);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PROJECT, PREFIX_DEADLINE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddTaskCommand.MESSAGE_USAGE));
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME);
```

Comment:
Good catch, have modified it on my local repo so that duplicate prefixes are not allowed for the project, task name or the deadline:

<img width="755" alt="image" src="https://github.com/AY2324S1-CS2103T-T08-3/tp/assets/64006665/6e9d2f10-ac31-4f11-8a56-6bb5c9ea6c5d">